core/primitives/src/stateless_validation/spice_chunk_endorsement.rs

```diff
@@ -0,0 +1,270 @@
+        })
+    }
+
+    fn to_signed_data(&self) -> SpiceEndorsementSignedData {
+        SpiceEndorsementSignedData {
+            execution_result_hash: self.execution_result.compute_hash(),
+            chunk_id: self.chunk_id.clone(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SpiceVerifiedEndorsement {
+    execution_result: ChunkExecutionResult,
+    chunk_id: SpiceChunkId,
+    account_id: AccountId,
+    signature: Signature,
+}
+
+impl SpiceVerifiedEndorsement {
+    pub fn account_id(&self) -> &AccountId {
+        &self.account_id
+    }
+
+    pub fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    pub fn chunk_id(&self) -> &SpiceChunkId {
+        &self.chunk_id
+    }
+
+    pub fn execution_result(&self) -> &ChunkExecutionResult {
+        &self.execution_result
+    }
+
+    pub fn to_stored(&self) -> SpiceStoredVerifiedEndorsement {
+        SpiceStoredVerifiedEndorsement {
+            execution_result_hash: self.execution_result().compute_hash(),
+            signature: self.signature.clone(),
+        }
+    }
+}
+
+/// Endorsement Core Statement that is included in blocks.
+/// This type intentionally has small amount to getters to make sure that callers verify signature
+/// before accessing endorsement data.
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, ProtocolSchema)]
+pub struct SpiceEndorsementCoreStatement {
+    account_id: AccountId,
+    signature: Signature,
+    signed_data: SpiceEndorsementSignedData,
+}
+
+impl SpiceEndorsementCoreStatement {
+    pub fn chunk_id(&self) -> &SpiceChunkId {
+        &self.signed_data.chunk_id
+    }
+
+    pub fn account_id(&self) -> &AccountId {
+        &self.account_id
+    }
+
+    pub fn verified_signed_data(
+        &self,
+        public_key: &PublicKey,
+    ) -> Option<(&SpiceEndorsementSignedData, &Signature)> {
+        let data = &self.signed_data.serialize_data_for_signing();
+        if !self.signature.verify(data, public_key) {
+            return None;
+        }
+        Some((&self.signed_data, &self.signature))
+    }
+
+    /// Converts to SpiceStoredVerifiedEndorsement without signature verification.
+    /// Caller should make sure that relevant core statement is validated.
+    pub fn unchecked_to_stored(&self) -> SpiceStoredVerifiedEndorsement {
+        SpiceStoredVerifiedEndorsement {
+            execution_result_hash: self.signed_data.execution_result_hash.clone(),
+            signature: self.signature.clone(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, ProtocolSchema, Hash)]
+pub struct SpiceEndorsementSignedData {
+    pub execution_result_hash: ChunkExecutionResultHash,
+    pub chunk_id: SpiceChunkId,
+}
+
+impl SpiceEndorsementSignedData {
+    fn serialize_data_for_signing(&self) -> Vec<u8> {
+        static SIGNATURE_DIFFERENTIATOR: StaticSignatureDifferentiator = "SpiceChunkEndorsement";
```

Comment:
I didn't initially realize that we need signature differentiator as well so added it just now in d8e4c96

I believe there is no need for it to travel over the wire, or exist at all outside of signing and checking signatures (even though for all other types currently I think we make it go over the wire with the rest of data).